source/config/__init__.py

```diff
@@ -1283,6 +1284,31 @@ def __setitem__(
 		self.manager._markWriteProfileDirty()
 		self._cache[key] = val
 
+		# Alias [documentFormatting][reportFontAttributes] for backwards compatibility.
+		# Todo: Remove in 2025.1.
+		if version_year < 2025and NVDAState._allowDeprecatedAPI():
+			self._link_reportFontAttributes_and_fontAttributeReporting(key, val)
+
+	def _link_reportFontAttributes_and_fontAttributeReporting(self, key, val):
+		if not (self.path == ("documentFormatting",)):
```

Comment:
May be consider having a match statement for self.path and self.key, so it can be expanded easily.

```Py
		match self.path:
			case ("documentFormatting",):
				match self.key:
					case "fontAttributeReporting":
						...
			case ("speech",):  # Needed for #16823 
```